src/basic/calendarspec.c

```diff
@@ -1014,7 +1014,7 @@ static bool tm_out_of_bounds(const struct tm *tm, bool utc) {
 
         t = *tm;
 
-        if (mktime_or_timegm(&t, utc) == (time_t) -1)
+        if (mktime_or_timegm(&t, utc) == (time_t) -1 || tm->tm_year > 2199)
```

Comment:
Can you move this threshold of 2199 somewhere to the top as a macro? Somewhere next to MAX_RANGE_LEN?

Also, I think this deserves to be a separate if check, with a quick explanatory comment why this check matters.